compiler/frontend/src/org/jetbrains/kotlin/diagnostics/PositioningStrategies.kt

```diff
@@ -745,19 +743,29 @@ object PositioningStrategies {
         }
     }
 
-    val REFERENCE_BY_QUALIFIED: PositioningStrategy<PsiElement> = object : PositioningStrategy<PsiElement>() {
+    val REFERENCE_BY_QUALIFIED: PositioningStrategy<PsiElement> = FindReferencePositioningStrategy(false)
+    val REFERENCED_NAME_BY_QUALIFIED: PositioningStrategy<PsiElement> = FindReferencePositioningStrategy(true)
+
+    class FindReferencePositioningStrategy(val locateReferencedName: Boolean) : PositioningStrategy<PsiElement>() {
         override fun mark(element: PsiElement): List<TextRange> {
-            when (element) {
+            var result = when (element) {
                 is KtQualifiedExpression -> {
                     when (val selectorExpression = element.selectorExpression) {
-                        is KtCallExpression -> return mark(selectorExpression.calleeExpression ?: selectorExpression)
-                        is KtReferenceExpression -> return mark(selectorExpression)
+                        is KtCallExpression -> selectorExpression.calleeExpression ?: selectorExpression
+                        is KtReferenceExpression -> selectorExpression
+                        else -> element
                     }
                 }
-                is KtCallExpression -> return mark(element.calleeExpression ?: element)
-                is KtConstructorDelegationCall -> return mark(element.calleeExpression ?: element)
+                is KtCallExpression -> element.calleeExpression ?: element
+                is KtConstructorDelegationCall -> element.calleeExpression ?: element
+                is KtOperationExpression -> element.operationReference
+                is KtWhenConditionInRange -> element.operationReference
```

Comment:
Just to double-check, these two new additions don't have `?: element` and is that intentional?

Reply:
Yeah it's intentional since the fields are marked `@NotNull`. Let me add a type to `result` to make this clear.